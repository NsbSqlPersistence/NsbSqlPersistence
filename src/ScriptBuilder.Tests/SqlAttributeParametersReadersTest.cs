using System.Collections.Generic;
using NUnit.Framework;
using ObjectApproval;

[TestFixture]
public class SqlAttributeParametersReadersTest
{

    [Test]
    public void Defaults()
    {
        var result = SettingsAttributeReader.Read(
            new CustomAttributeMock(
                new Dictionary<string, object>
                {
                    {
                        //At least one is required
                        "MsSqlServerScripts", true
                    }
                }));
        ObjectApprover.VerifyWithJson(result);
    }

    [Test]
    public void NonDefaults()
    {
        var result = SettingsAttributeReader.Read(
            new CustomAttributeMock(
                new Dictionary<string, object>
                {
                    {
                        "ScriptPromotionPath", @"D:\scripts"
                    },
                    {
                        "MsSqlServerScripts", true
                    },
                    {
                        "MySqlScripts", true
                    },
                    {
                        "OracleScripts", true
                    },
                    {
                        "ProduceSagaScripts", false
                    },
                    {
                        "ProduceTimeoutScripts", false
                    },
                    {
                        "ProduceSubscriptionScripts", false
                    },
                    {
                        "ProduceOutboxScripts", false
                    }
                }));
        ObjectApprover.VerifyWithJson(result);
    }


}