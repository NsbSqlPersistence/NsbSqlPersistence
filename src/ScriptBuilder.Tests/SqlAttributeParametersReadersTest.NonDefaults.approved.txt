{
  "BuildVariants": [
    "MsSqlServer",
    "MySql",
    "Oracle"
  ],
  "ScriptPromotionPath": "D:\\scripts",
  "ProduceOutboxScripts": false,
  "ProduceSubscriptionScripts": false,
  "ProduceTimeoutScripts": false,
  "ProduceSagaScripts": false
}